src/components/select/SelectWithFilter.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {ReduxConnect} from '../../utils/ReduxUtils';
+import {FilterBoxConnected} from '../filterBox/FilterBoxConnected';
+import {IItemBoxProps} from '../itemBox/ItemBox';
+import {ISelectProps} from './SelectConnected';
+
+export interface ISelectWithFilterProps extends ISelectProps {
+  matchFilter?: (filterValue: string, item: IItemBoxProps) => boolean;
+}
+
+export const selectWithFilter = (Component: (React.ComponentClass<ISelectWithFilterProps> | React.StatelessComponent<ISelectWithFilterProps>)): React.ComponentClass<ISelectWithFilterProps> => {
+  const defaultMatchFilter = (filterValue: string, item: IItemBoxProps) => {
+    if (filterValue === '') {
+      return true;
+    }
+
+    const regex = new RegExp(filterValue, 'gi');
```

Comment:
pas besoin du `new` IIRC

Reply:
Since we create an object instance I think it's better to use the new